plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/functions/tablechanges/TableChangesFunction.java

```diff
@@ -0,0 +1,186 @@
+import io.trino.spi.function.table.TableFunctionAnalysis;
+import io.trino.spi.type.TypeManager;
+import io.trino.spi.type.VarcharType;
+import org.apache.iceberg.SchemaParser;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableProperties;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.trino.plugin.iceberg.ColumnIdentity.TypeCategory.PRIMITIVE;
+import static io.trino.plugin.iceberg.IcebergColumnHandle.DATA_CHANGE_ORDINAL_ID;
+import static io.trino.plugin.iceberg.IcebergColumnHandle.DATA_CHANGE_ORDINAL_NAME;
+import static io.trino.plugin.iceberg.IcebergColumnHandle.DATA_CHANGE_TIMESTAMP_ID;
+import static io.trino.plugin.iceberg.IcebergColumnHandle.DATA_CHANGE_TIMESTAMP_NAME;
+import static io.trino.plugin.iceberg.IcebergColumnHandle.DATA_CHANGE_TYPE_ID;
+import static io.trino.plugin.iceberg.IcebergColumnHandle.DATA_CHANGE_TYPE_NAME;
+import static io.trino.plugin.iceberg.IcebergColumnHandle.DATA_CHANGE_VERSION_ID;
+import static io.trino.plugin.iceberg.IcebergColumnHandle.DATA_CHANGE_VERSION_NAME;
+import static io.trino.plugin.iceberg.TypeConverter.toTrinoType;
+import static io.trino.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static io.trino.spi.function.table.ReturnTypeSpecification.GenericTable.GENERIC_TABLE;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS;
+import static java.util.Objects.requireNonNull;
+
+public class TableChangesFunction
+        extends AbstractConnectorTableFunction
+{
+    private static final String FUNCTION_NAME = "table_changes";
+    private static final String SCHEMA_VAR_NAME = "SCHEMA";
+    private static final String TABLE_VAR_NAME = "TABLE";
+    private static final String START_SNAPSHOT_VAR_NAME = "START_SNAPSHOT_ID";
+    private static final String END_SNAPSHOT_VAR_NAME = "END_SNAPSHOT_ID";
+
+    private final TrinoCatalogFactory trinoCatalogFactory;
+    private final TypeManager typeManager;
+
+    @Inject
+    public TableChangesFunction(TrinoCatalogFactory trinoCatalogFactory, TypeManager typeManager)
+    {
+        super(
+                "system",
+                FUNCTION_NAME,
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder()
+                                .name(SCHEMA_VAR_NAME)
+                                .type(VarcharType.createUnboundedVarcharType())
+                                .build(),
+                        ScalarArgumentSpecification.builder()
+                                .name(TABLE_VAR_NAME)
+                                .type(VarcharType.createUnboundedVarcharType())
+                                .build(),
+                        ScalarArgumentSpecification.builder()
```

Comment:
Would it be possible for now to make start_snapshot_version default to first version and thus be optional? 
Can we also get rid of last version for the first iteration ? 
Also what would happen if table schema evolves between read versions? I think it should be handles correctly but Im not sure